WordPress/gutenberg
# Table Block: Limit the max number of row/column fields when creating the table

Fix #44808

## What?
This PR limits the max number of row/column fields to **30** when creating tables.

This limitation is only valid when creating a table, and once the table is created, the number of rows and columns can be increased beyond the limit.

## Why?
Currently, there is no limit when creating rows/columns, which causes the block editor to freeze or causes out-of-memory errors in the browser when huge values are specified.

## How?
The upper limit would depend on the environment, but I specified 30. I felt that this was an acceptable number for the time it takes for the table to be rendered and the response to entering cell text is just barely acceptable.

Examples of a block that limits the maximum number of:

- The columns block can be specified up to 6 columns in the sidebar, but more columns can be inserted by the block inserter.
- The tag cloud block limits the maximum number of tags to 100.
- The page list block within the navigation block allows being converted to a page link only if the number of pages is within 100.

## Testing Instructions

- Insert a table block.
- Enter a number greater than 30 in the row or column field.
- Confirm that the value is converted to 30.
